### PR TITLE
Allow deletion beyond last 100 packages

### PR DESCRIPTION
**Pull request by NamrataJha**
This PR adds pagination in the `delete-package-versions` action and extend its functionality beyond the last 100 package versions.

### Changes in the PR
- Pagination enabled for get-versions graphQL calls.
- MAX_LIMIT = 99 introduced on no of packages to be deleted in 1 run. This was not mentioned in the ReadMe earlier but the action was failing on reaching this limit. Has been added to ReadME now.
- Throw error when action configured with invalid values.
- Action output will only print the no of versions deleted till that time not all the deleted version ids.
- Add in ReadME that this action does not supports ghcr packages

### What to Test
- Test using the latest commit of delete-package-versions action - `actions/delete-package-versions@aac34a9f3354ee3c26e304066740d686bee4c5f5` to use the changes.
- Configure the workflow with valid combinations as given in ReadME to run for packages with more than 100 versions.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -42,14 +42,12 @@ This action deletes versions of a package from [GitHub Packages](https://github.
...
   # Defaults to 1.
-  # Cannot be more than 100.
   num-old-versions-to-delete:
 
   # The number of latest versions to not delete.
-  # Defaults to 0.
-  # When this is set greater than 0 it will delete all deletable package versions except the specified no.
-  # `num-old-versions-to-delete` will not be taken into account with this option.
-  # Cannot be more than 100.
+  # This cannot be specified with `num-old-versions-to-delete`. By default, `num-old-versions-to-delete` is taken into account.
```
Comment (nadesu) [inline README.md:46]: Call it as This cannot be specified with `num-old-versions-to-delete`. By default,`num-old-versions-to-delete` takes precedence over `min-versions-to-keep`

File: src/delete.ts [hunk 2]
```diff
@@ -1,46 +1,130 @@
...
+          Here first filter out the versions that are to be ignored.
+          Then update input.numOldeVersionsToDelete to the no of versions deleted from the next 100 versions batch.
+          */
+          value = value.filter(info => !input.ignoreVersions.test(info.version))
+          const temp = input.numOldVersionsToDelete
+          input.numOldVersionsToDelete =
+            input.numOldVersionsToDelete - value.length <= 0
+              ? 0
+              : input.numOldVersionsToDelete - value.length
+          return value.map(info => info.id).slice(0, temp)
```
[the remark this answers is not in the document]
Comment (NamrataJha) [reply src/delete.ts:76]: We are slicing the value array only so using value.length will always return the complete array. If temp > value.length, value.length elements will be returned only as there wont be any extra elements

[the remark this answers is not in the document]
Comment (NamrataJha) [reply src/delete.ts:76]: This temp in terms of slicing will only make a difference when suppose we have 25 versions after filtering but only want to delete 10

File: src/delete.ts [hunk 3]
```diff
@@ -49,14 +121,24 @@ export function deleteVersions(input: Input): Observable<boolean> {
     return throwError('No token found')
   }
 
-  if (input.numOldVersionsToDelete <= 0) {
+  if (!input.checkInput()) {
+    return throwError('Invlaid input combination')
```
[the remark this answers is not in the document]
Comment (NamrataJha) [reply src/delete.ts:137]: Updating this.

**Review by AmrutaKawade — APPROVED** [review]

**Pull request merged**